test/e2e/prometheus_test.go

```diff
@@ -94,152 +90,178 @@ func TestAntiAffinity(t *testing.T) {
+					expected:    func(v int) bool { return v != 0 },
+					description: "prometheus_remote_storage_samples_pending indicates no remote write progress, expected a continuously changing delta",
+				},
+			},
 		},
 		{
-			name: "assert mtls remote write works",
-			port: "8081",
+			name: "assert remote write with mtls authorization works",
 			rwSpec: `
-  - url: https://%s
+  - url: https://%[1]s/api/v1/write
     tlsConfig:
       ca:
         secret:
-          name: selfsigned-mtls-bundle
+          name: %[2]s
           key: ca.crt
       cert:
         secret:
-          name: selfsigned-mtls-bundle
+          name: %[2]s
           key: client.crt
       keySecret:
-        name: selfsigned-mtls-bundle
+        name: %[2]s
         key: client.key
 `,
+			expected: []remoteWriteTest{
+				{
+					query:       fmt.Sprintf(`ceil(delta(prometheus_remote_storage_samples_pending{pod="%[1]s",prometheus_replica="%[1]s"}[1m]))`, "prometheus-k8s-0"),
+					expected:    func(v int) bool { return v != 0 },
+					description: "prometheus_remote_storage_samples_pending indicates no remote write progress, expected a continuously changing delta",
+				},
+				{
+					query:       fmt.Sprintf(`ceil(delta(prometheus_remote_storage_samples_pending{pod="%[1]s",prometheus_replica="%[1]s"}[1m]))`, "prometheus-k8s-1"),
+					expected:    func(v int) bool { return v != 0 },
+					description: "prometheus_remote_storage_samples_pending indicates no remote write progress, expected a continuously changing delta",
+				},
+			},
+		},
+		{
+			name: "assert remote write temporary cluster id label is dropped",
+			rwSpec: `
+  - url: https://%[1]s/api/v1/write
+    tlsConfig:
+      ca:
+        secret:
+          name: %[2]s
+          key: ca.crt
+`,
+			expected: []remoteWriteTest{
+				{
+					query:       `absent(prometheus_remote_storage_samples_pending{__tmp_openshift_cluster_id__=~".+"})`,
+					expected:    func(v int) bool { return v == 1 },
+					description: "Expected to find 0 time series of metric prometheus_remote_storage_samples_pending with the temporary cluster_id label",
+				},
+			},
+		},
+		{
+			name: "assert remote write cluster_id relabel config works",
+			rwSpec: `
+  - url: https://%[1]s/api/v1/write
+    tlsConfig:
+      ca:
+        secret:
+          name: %[2]s
+          key: ca.crt
+    writeRelabelConfigs:
+    - sourceLabels:
+        - __tmp_openshift_cluster_id__
+      targetLabel: cluster_id
+      action: replace
+`,
+			expected: []remoteWriteTest{
+				{
+					query:       `count(prometheus_remote_storage_samples_pending{cluster_id!=""})`,
+					expected:    func(v int) bool { return v == 4 },
+					description: "Expected to find 4 time series of metric prometheus_remote_storage_samples_pending with the cluster_id label",
+				},
+				{
+					query:       `absent(prometheus_remote_storage_samples_pending{__tmp_openshift_cluster_id__=~".+"})`,
```

Comment:
I think you might have forgotten to update this, no?

Reply:
I'm afraid I need more hints :) 
The description looks a bit weird but is actually correct, if that is what you mean. 
> absent(v instant-vector) returns an empty vector if the vector passed to it has any elements and a 1-element vector with the value 1 if the vector passed to it has no elements.

Reply:
Sorry, my mistake! Please ignore.